solara/toestand.py

```diff
@@ -97,6 +97,35 @@ def __init__(self, merge: Callable = merge_state, equals=equals_extra):
         self.equals = equals
         self.listeners: Dict[str, Set[Tuple[Callable[[T], None], Optional[ContextManager]]]] = defaultdict(set)
         self.listeners2: Dict[str, Set[Tuple[Callable[[T, T], None], Optional[ContextManager]]]] = defaultdict(set)
+        self._prevent_boolean = solara.settings.main.prevent_reactive_boolean
+
+    # make sure all boolean operations give type errors
+    def __bool__(self):
+        if self._prevent_boolean:
+            raise TypeError("Reactive vars are not allowed in boolean expressions, did you mean to use .value?")
+        return True
```

Comment:
```suggestion
    if solara.settings.main.prevent_reactive_boolean:
      def __bool__(self):
          if self._prevent_boolean:
              raise TypeError("Reactive vars are not allowed in boolean expressions, did you mean to use .value?")
          return True
```

This may look funny at first, but that does not require up to give a default implementation. The other solution would be to call super(self).__bool__()